### PR TITLE
Add recursive title lookup and html builder for debugging

### PR DESCRIPTION
**Pull request by Ymil**
Hi People! Thanks for create official fork :D
I made some changes so that the tool works well with my project, I hope it helps you.

Add recursion in the title search because information was lost when using nested titles.

In this image I leave an example of nested titles.
imagen

And here, on the left you can see what happens with the original version and on the right with the recursion.

imagen
-------------
I added the HTML generation because it's easier for me when it comes to editing the HTML.

I'm waiting for your answer, cheers.!

### CODE REVIEW

File: mkpdfs_mkdocs/generator.py [hunk 1]
```diff
@@ -152,7 +159,7 @@ def add_tocs(self):
             h3 = self.html.new_tag('h3')
             h3.insert(0, n.title)
             self._toc.append(h3)
-            if n.is_page:
+            if n.is_page and len(n.toc) > 0:
```
Comment (jurgenwigg) [inline mkpdfs_mkdocs/generator.py:162]:
```suggestion
            if n.is_page and n.toc:
```

**Review by jurgenwigg — CHANGES_REQUESTED** [review]
Sorry for so late review, but GitHub (or me) didn't send me the notice :) Cosmetic change and we are ready to go

**Comment by Ymil** [comment]
Brilliant! Thanks for the suggestion, it's already applied!

**Comment by jurgenwigg** [comment]
Great!

**Pull request merged**